react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+
+class Home extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      images: [
+        { photoId: 0, url: 'images/001.png' },
+        { photoId: 1, url: 'images/004.png' },
+        { photoId: 2, url: 'images/007.png' },
+        { photoId: 3, url: 'images/025.png' },
+        { photoId: 4, url: 'images/039.png' }
+      ],
+      currentPhoto: 0
+    };
+    this.circleClick = this.circleClick.bind(this);
+    this.arrowClick = this.arrowClick.bind(this);
+  }
+
+  componentDidMount() {
+    this.carouselInterval = setInterval(() => {
+      if (this.state.currentPhoto === 4) {
+        this.setState({ currentPhoto: -1 });
+      }
+      this.setState({ currentPhoto: this.state.currentPhoto + 1 });
+    }, 3000);
+  }
+
+  circleClick(event) {
+    clearInterval(this.carouselInterval);
+    this.setState({
+      currentPhoto: parseInt(event.target.getAttribute('imageindex'))
+    });
+    this.componentDidMount();
+  }
+
+  arrowClick(event) {
+
+    clearInterval(this.carouselInterval);
+    if (event.target.className === 'arrow left relative-fifty previous-arrow' && this.state.currentPhoto === 0) {
```

Comment:
We don't modify elements like this in React, instead pass props to elements in `render()` to change them.